  @Test
  public void getAllResults() throws Exception {
    List<Integer> expectedValues = Arrays.asList(1, 2, 3, 4, 5, 6, 7);

    CompletableFuture<List<Integer>> results = Async.waitForAllSuccessfulResponses(
        expectedValues.stream()
            .map(CompletableFuture::completedFuture)
            .collect(Collectors.toList()));

    List<Integer> actualValues = results.get();

    Collections.sort(expectedValues);
    Collections.sort(actualValues);

    assertEquals(expectedValues, actualValues);
  }

